lib/ansible/modules/network/junos/junos_facts.py

```diff
@@ -86,316 +101,28 @@
-                if 'object' in value:
-                    value['object'] = dict(value['object'])
-
-        return facts
-
-
-FACT_SUBSETS = dict(
-    default=Default,
-    hardware=Hardware,
-    config=Config,
-    interfaces=Interfaces,
-    ofacts=OFacts
-)
-
-VALID_SUBSETS = frozenset(FACT_SUBSETS.keys())
+from ansible.module_utils.network.junos.argspec.facts.facts import FactsArgs
+from ansible.module_utils.network.junos.facts.facts import Facts
+from ansible.module_utils.network.junos.junos import junos_argument_spec
 
 
 def main():
     """ Main entry point for AnsibleModule
     """
-    argument_spec = dict(
-        gather_subset=dict(default=['!config', '!ofacts'], type='list'),
-        config_format=dict(default='text', choices=['xml', 'text', 'set', 'json']),
-    )
-
+    argument_spec = FactsArgs.argument_spec
     argument_spec.update(junos_argument_spec)
 
     module = AnsibleModule(argument_spec=argument_spec,
                            supports_check_mode=True)
 
-    warnings = list()
-    gather_subset = module.params['gather_subset']
-
-    runable_subsets = set()
-    exclude_subsets = set()
-
-    for subset in gather_subset:
-        if subset == 'all':
-            runable_subsets.update(VALID_SUBSETS)
-            continue
-
-        if subset.startswith('!'):
-            subset = subset[1:]
-            if subset == 'all':
-                exclude_subsets.update(VALID_SUBSETS)
-                continue
-            exclude = True
-        else:
-            exclude = False
-
-        if subset not in VALID_SUBSETS:
-            module.fail_json(msg='Subset must be one of [%s], got %s' %
-                             (', '.join(sorted([subset for subset in
-                                                VALID_SUBSETS])), subset))
-
-        if exclude:
-            exclude_subsets.add(subset)
-        else:
-            runable_subsets.add(subset)
-
-    if not runable_subsets:
-        runable_subsets.update(VALID_SUBSETS)
-
-    runable_subsets.difference_update(exclude_subsets)
-    runable_subsets.add('default')
-
-    # handle fetching old style facts separately
-    runable_subsets.discard('ofacts')
-
-    facts = dict()
-    facts['gather_subset'] = list(runable_subsets)
-
-    instances = list()
-    ansible_facts = dict()
-
-    # fetch old style facts only when explicitly mentioned in gather_subset option
-    if 'ofacts' in gather_subset:
-        if HAS_PYEZ:
-            ansible_facts.update(OFacts(module).populate())
-        else:
-            warnings += ['junos-eznc is required to gather old style facts but does not appear to be installed. '
-                         'It can be installed using `pip  install junos-eznc`']
-        facts['gather_subset'].append('ofacts')
-
-    for key in runable_subsets:
-        instances.append(FACT_SUBSETS[key](module))
+    warnings = ['default value for `gather_subset` '
+                'will be changed to `min` from `!config` v2.11 onwards']
```

Comment:
`min` isn't listed currently as one of the possible values for `gather_subset`. Is that going to be added in 2.11?

Reply:
Yes that's right